### PR TITLE
Update to markdownlint v0.22.0

### PR DESCRIPTION
**Pull request by igetgames**
👋🏽 Hi! I've been using your extension for a while and I'm happy to contribute something back.

Separate from the markdownlint v0.22.0 upgrade, I added an `.editorconfig` to reflect the indent style and trailing whitespace rules used in the project. I also added `package-lock.json` and `yarn.lock` to `.gitignore` to make it easier to drop into development.

### CODE REVIEW

**Comment by DavidAnson** [comment]
The updates for v0.22.0 look great, thank you! I think you caught all the places which was a bit harder for this release in particular.

Regarding `.editorconfig`, I don't use anything that supports it (I don't think) and I'd prefer not to have files that aren't used lying around. If there are lint rules to enforce these policies, that's my preference (and I think I'm already using what's available in that respect).

Regarding lock files, I would prefer a project setting that prevented them from being created at all because they have no meaning for this project due to how extensions are published. Absent that, I would be afraid that ignoring them as you propose would lead to them getting accidentally bundled or included with a release.

**Comment by igetgames** [comment]
Thanks for the quick feedback. I've updated the PR to remove `.editorconfig` and took your suggestion to ignore lockfiles, adding them to `.vscodeignore`. I also disabled lockfile generation from npm, but couldn't find an equivalent Yarn feature.

**Review by DavidAnson — CHANGES_REQUESTED** [review]
The lock file changes are cute, but feel a little like chasing a tail. There are two more files now, but they still don't solve the whole problem. How about let's just make the version update changes in this PR. What I see there seems good and uncontroversial. Regarding updates to the schema, could you please run `npm run schema` if you didn't already? It'll pick up any missing changes. Thanks!

File: README.md [hunk 1]
```diff
@@ -294,6 +293,7 @@ The following snippets are available to help (press `Ctrl+Space` for IntelliSens
 * `markdownlint-disable-file`
 * `markdownlint-enable-file`
 * `markdownlint-configure-file`
+* `markdownlint-disable-next-line`
```
Comment (DavidAnson) [inline README.md:296]: Nitpick: Please move this after `markdownlint-enable` to match the order at https://github.com/DavidAnson/markdownlint#configuration

**Review by DavidAnson — CHANGES_REQUESTED** [review]

**Comment by igetgames** [comment]
Sure. I've dropped the lockfile changes, and made the requested change to `README.md`. I had run `npm run schema` while working on the other changes, it is what had picked up the punctuation changes.

**Review by DavidAnson — APPROVED** [review]
Looks good!

**Pull request merged**